components/xwiki-platform-users/xwiki-platform-users-wiki/src/main/java/org/xwiki/users/internal/WikiUser.java

```diff
@@ -94,14 +94,15 @@ public String getName()
         if (this.profileReference == null) {
             return "";
         }
-        String result = this.bridge.getProperty(this.profileReference, this.classReference, "first_name") + "";
-        result = StringUtils.trim(result) + " ";
-        result +=
-            StringUtils.trim("" + this.bridge.getProperty(this.profileReference, this.classReference, "last_name"));
+        String firstName =
+            String.valueOf(this.bridge.getProperty(this.profileReference, this.classReference, "first_name"));
+        String lastName =
+            String.valueOf(this.bridge.getProperty(this.profileReference, this.classReference, "last_name"));
+        String result = StringUtils.trim(firstName) + " " + StringUtils.trim(lastName);
         if (StringUtils.isBlank(result)) {
             result = getUsername();
         }
-        return result.trim();
```

Comment:
This might still be needed for example when firstname is set, but lastname is empty. This will show up as "firstname ".